src/cpu/aarch64/gemm_arm_convolution.hpp

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+* Copyright 2020 Arm Ltd. and affiliates
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_GEMM_ARM_CONVOLUTION_HPP
+#define CPU_GEMM_ARM_CONVOLUTION_HPP
+
+#include "common/c_types_map.hpp"
+#include "common/memory_tracking.hpp"
+#include "common/primitive.hpp"
+
+#include "cpu/aarch64/gemm_arm_convolution_utils.hpp"
+#include "cpu/gemm/gemm.hpp"
+#include "cpu/ref_eltwise.hpp"
+
+#include "cpu/cpu_convolution_pd.hpp"
+
+#include "arm_compute/runtime/NEON/NEFunctions.h"
+#include "arm_compute/runtime/Scheduler.h"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+struct acl_obj_t {
+    arm_compute::NEGEMMConvolutionLayer gemm_conv;
+    arm_compute::Tensor src_tensor;
+    arm_compute::Tensor wei_tensor;
+    arm_compute::Tensor bia_tensor;
+    void *bia_memory;
+    arm_compute::Tensor dst_tensor;
+};
+
+struct acl_resource_t : public resource_t {
+    acl_resource_t() : acl_obj_(utils::make_unique<acl_obj_t>()) {}
```

Comment:
So the reason I specified the constructor was to avoid initializing any memory in the resource configure() function in an attempt to be more respectful of the C++ RAII model. So instead of using .reset() on the unique_ptr in configure() I am only checking that the memory is valid.

If I'm misunderstanding this I'm happy to go back to the .reset() alternative.

Reply:
Yeah, I misunderstood the original code.
I thought that 
``` cpp
acl_resource_t() : acl_obj_(utils::make_unique<acl_obj_t>()) {}
```
is the same as 
``` cpp
acl_resource_t() = default;
```

However, the difference is that in the former (and the current state of the PR) we create a default `acl_obj_t` object, while in the latter case we will need to allocate it at `configure()`. I don't think there is significant difference between those, so I am with the current approach. @dzarukin, what do you think?

Reply:
Assume resolved :)